apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,132 @@
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        user_id: req.body.student_uid,
+        student_uid: req.body.student_uid,
+        group_id: null,
+      };
+      // First, validate if group belongs to the assessment
+      if (params.group_name) {
+        const group_result = await sqldb.queryZeroOrOneRowAsync(sql.select_group_in_assessment, {
+          group_name: params.group_name,
+          course_instance_id: res.locals.course_instance.id,
+          assessment_id: res.locals.assessment.id,
+        });
+        // Get the group_id from the result
+        if (group_result.rows.length > 0) {
+          params.group_id = group_result.rows[0].id;
+          console.log('GROUP ID: ' + params.group_id);
+        } else {
+          params.group_id = null;
+        }
+
+        // If group does not belong to assessments and indirectly course instances, return error
+        if (!params.group_id) {
+          throw error.make(400, 'Group not found in this assessment.');
+        }
+      } else if (params.user_id) {
+        const user = await selectUserByUid(params.user_id);
+        if (!user) {
+          throw error.make(400, 'Student UID does not exist.');
+        }
+        params.user_id = user.user_id;
+
+        const enrollment = await getEnrollmentForUserInCourseInstance({
+          user_id: params.user_id,
+          course_instance_id: res.locals.course_instance.id,
+        });
+        if (!enrollment) {
+          throw error.make(400, 'Student is not enrolled in the current course instance.');
+        }
+      }
+      await sqldb.queryAsync(sql.insert_assessment_access_policy, params);
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'delete_override') {
+      await sqldb.queryAsync(sql.delete_assessment_access_policy, {
+        unsafe_assessment_access_policies_id: req.body.policy_id,
+      });
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'edit_override') {
+      const edit_params = {
+        assessment_id: res.locals.assessment.id,
+        credit: req.body.credit,
+        end_date: new Date(req.body.end_date),
+        group_name: req.body.group_name || null,
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        user_id: req.body.student_uid,
+        group_id: null,
+        student_uid: req.body.student_uid || null,
+      };
+      const user = await selectUserByUid(edit_params.user_id);
+      if (!user) {
+        throw error.make(400, 'Student UID does not exist.');
+      }
+      const enrollment = await getEnrollmentForUserInCourseInstance({
+        user_id: user.user_id,
+        course_instance_id: res.locals.course_instance.id,
+      });
+
+      if (!enrollment) {
+        throw error.make(400, 'Student is not enrolled in the current course instance.');
+      }
+      edit_params.user_id = user.user_id;
```

Comment:
This branch runs even for group assessments. Is that desired? What happens when it _does_ run in that case?

Reply:
added to conditional in 6d6e070278f2e22de9aacb64f3e9d975bcc47983